openmc/model/model.py

```diff
@@ -1154,51 +1175,85 @@ def update_material_volumes(self, names_or_ids, volume):
 
         self._change_py_lib_attribs(names_or_ids, volume, 'material', 'volume')
 
-    def differentiate_depletable_mats(self, diff_volume_method: str):
+    def differentiate_depletable_mats(self, diff_volume_method : str = None):
         """Assign distribmats for each depletable material
 
         .. versionadded:: 0.14.0
 
+        .. version added:: 0.15.1-dev
+            diff_volume_method default is None, do not apply volume to the new
+            materials. Is now a convenience method for
+            differentiate_mats(diff_volume_method, depletable_only=True)
+
         Parameters
         ----------
         diff_volume_method : str
             Specifies how the volumes of the new materials should be found.
-            Default is to 'divide equally' which divides the original material
-            volume equally between the new materials, 'match cell' sets the
-            volume of the material to volume of the cell they fill.
+            Default is to 'None', do not apply volume to the new materials,
+            'divide equally' which divides the original material
+            volume equally between the new materials,
+            'match cell' sets the volume of the material to volume of the cell
+            they fill.
         """
+        self.differentiate_mats(diff_volume_method, depletable_only=True)
+
+    def differentiate_mats(self, diff_volume_method: str = None, depletable_only: bool = True):
+        """Assign distribmats for each material
+
+        .. versionadded:: 0.15.1-dev
+
+        Parameters
+        ----------
+        diff_volume_method : str
+            Specifies how the volumes of the new materials should be found.
+            Default is to 'None', do not apply volume to the new materials, 
+            'divide equally' which divides the original material
+            volume equally between the new materials, 
+            'match cell' sets the volume of the material to volume of the cell 
+            they fill.
+        depletable_only : bool
+            Default is True, only depletable materials will be differentiated all materials will be 
+            differentiated otherwise.
+        """
+        check_value('volume differentiation method', diff_volume_method, ["divide equally", "match cell", None])
+
         # Count the number of instances for each cell and material
         self.geometry.determine_paths(instances_only=True)
 
-        # Extract all depletable materials which have multiple instances
+        # Extract all or depletable_only materials which have multiple instance
         distribmats = set(
             [mat for mat in self.materials
-                if mat.depletable and mat.num_instances > 1])
+                if (mat.depletable or not depletable_only) and mat.num_instances > 1])
 
-        if diff_volume_method == 'divide equally':
+        if diff_volume_method == "divide equally":
             for mat in distribmats:
                 if mat.volume is None:
```

Comment:
Let's move this check into it's own loop over `distribmats`. If error it hit for one material but not all, this will result in some Materials with volumes changed after an early exit, leaving their volume property in an invalid state.